spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -146,10 +145,10 @@ private void setCredentialsProperty() {
 		try {
 			// First tries the SQL configuration credential.
 			if (this.gcpCloudSqlProperties != null
-					&& this.gcpCloudSqlProperties.getCredentialsLocation() != null
-					&& Files.exists(this.gcpCloudSqlProperties.getCredentialsLocation())) {
+					&& this.gcpCloudSqlProperties.getCredentials() != null) {
 				System.setProperty(SqlCredentialFactory.CREDENTIAL_LOCATION_PROPERTY_NAME,
-						this.gcpCloudSqlProperties.getCredentialsLocation().toString());
+						this.gcpCloudSqlProperties.getCredentials().getLocation().getFile()
+								.getAbsolutePath());
```

Comment:
How about if `Resource` not file based?
```
public File getFile() throws IOException {
	throw new UnsupportedOperationException(
			getDescription() + " cannot be resolved to absolute file path");
}
```
for example.

Maybe we should have some `if...else` or `try...catch` and don't fail the whole application with the `FileNotFoundException`. Any way this looks like just a system property population...

Reply:
`FileNotFoundException` is an `IOException`, there is already a catch block for those.

Reply:
Ah! Good. I see.
Well, IMO that is bad practice to have such a big `try...catch` block. It is pretty easy to lose what is really wrong.
If we are indeed fine with the failure on `FileNotFound` would be better to have that catch on particular peace of code, around `this.gcpCloudSqlProperties.getCredentials().getLocation().getFile()`.

More over I see we have a logic a bit below like:
```
// This should happen if the Spring resource isn't in the filesystem, but a URL,
 				// classpath file, etc.
 				if (credentialsLocationFile == null) {
 					if (LOGGER.isWarnEnabled()) {
 						LOGGER.warn("The private key of the Google Cloud SQL credential must "
 								+ "be in a file on the filesystem.");
}
```
That's great message for end-developer, but it isn't reachable in case of non-file `Resource`.

Reply:
Moved things around. Let me know if you like it